pandas/io/json/_normalize.py

```diff
@@ -228,14 +228,17 @@ def json_normalize(
     Returns normalized data with columns prefixed with the given string.
     """
 
-    def _pull_field(js, spec):
-        result = js
+    def _pull_field(js: Dict[str, Any], spec: Union[List, str]) -> Iterable:
+        result: Any = js
         if isinstance(spec, list):
             for field in spec:
                 result = result[field]
         else:
             result = result[spec]
 
+        if not isinstance(result, Iterable):
```

Comment:
Should this just be `pd.isnull(result)`? The consequence of this being an isinstance check would be that all scalar values would return nothing right? 

I know there was some conversation around this before but I think should disambiguate between missing values and scalar values; the former is fine whereas the latter should raise a `TypeError`

Reply:
You have a point. But should the `TypeError` be raised here then to make it more user friendly? Because otherwise the `TypeError` will be raised from 

```
                recs = [
                    nested_to_record(r, sep=sep, max_level=max_level)
                    if isinstance(r, dict)
                    else r
                    for r in recs
                ]
```

Which is not very indicative  of what is wrong and more importantly where It took me quite some time to find out in my data what went wrong and why as it can just happen in the middle of a 100mb file.

Reply:
Yea I think can raise a TypeError here if not missing or Dict/List

Reply:
so I will do a check with `pd.isnull` and will check with `is_scalar`.

Reply:
Yea I think that would be better

Reply:
*sigh* `str` is defined as a scalar. I'm not seeing away around to do `is_scalar(result) and not isinstance(result, Iterable)`. Or is there a better way?

Reply:
`if is_scalar(result) and pd.isnull(result)`?

Reply:
when `pd_isnull(result) == true` I need to `result = []`. `pd_isnull` doesn't deal with Iterables :-). I have updated the PR with a suggestion.